internal/bminventory/inventory.go

```diff
@@ -340,9 +341,51 @@ func (b *bareMetalInventory) setDefaultRegisterClusterParams(ctx context.Context
 		}
 	}
 
+	params.NewClusterParams.NetworkType, err = getDefaultNetworkType(params)
+	if err != nil {
+		return params, err
+	}
+
 	return params, nil
 }
 
+// If the cluster is SNO, or the IP version is IPv6 (based on ClusterNetworkCidr param), or the OpenShift version >= 4.12,
```

Comment:
in the comment, you wrote that you are checking if "the IP version is IPv6" but I don't see you handle IPv6 in the method. 
I'm not sure what should be the fix: Add check for IPv6 or remove it from the comment

Reply:
Mistake, was corrected now